### PR TITLE
Fix casing of compile items in System.Speech and add CI protection

### LINKED ISSUE #50060: Building with allConfigurations on Linux fails on System.Speech
When I build on Linux with:

```shell
./build.sh -s clr+libs -rc release -allConfigurations
```

I get 12 build errors, all regarding System.Speech:

> CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/SrgsCompiler/SrgsCompiler.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]

<details>
<summary>All errors</summary>
<pre>
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/SrgsCompiler/SrgsCompiler.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/ISsmlParser.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/SsmlParser.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/TtsEngineProxy.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/TtsEvent.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/TtsVoice.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/SrgsCompiler/SrgsCompiler.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/ISsmlParser.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/SsmlParser.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/TtsEngineProxy.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/TtsEvent.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
CSC : error CS2001: Source file '/code/personal/dotnet/runtime/src/libraries/System.Speech/src/Internal/Synthesis/TtsVoice.cs' could not be found. [/code/personal/dotnet/runtime/src/libraries/System.Speech/src/System.Speech.csproj]
    0 Warning(s)
    12 Error(s)
</pre>
</details>

Is this expected?

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/50060

### CODE REVIEW

File: eng/pipelines/global-build.yml [hunk 1]
```diff
@@ -92,6 +92,21 @@ jobs:
...
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/common/global-build-job.yml
+    buildConfig: debug
+    platforms:
+    - Linux_x64
+    jobParameters:
+      nameSuffix: Libraries_AllConfigurations
+      buildArgs: -subset libs -allconfigurations
+      timeoutInMinutes: 90
```
Comment (stephentoub) [inline eng/pipelines/global-build.yml:108]: I agree this is desirable, but any idea what it'll do to pressure on CI?

Comment (ViktorHofer) [reply eng/pipelines/global-build.yml:108]: As this skips building runtimes, hosts and other components that are built in the other dev-innerloop legs, this leg won't stand out and take more time than the others. From an overall resource perspective, this adds one more leg to the already existing 150+ legs.

File: eng/pipelines/global-build.yml [hunk 2]
```diff
@@ -92,6 +92,21 @@ jobs:
       buildArgs: -subset mono+libs /p:RuntimeFlavor=Mono
       timeoutInMinutes: 90
 
+#
+# Build Libraries AllConfigurations. This excercises the code path where we build libraries for all
```
Comment (danmoseley) [inline eng/pipelines/global-build.yml:96]:
```suggestion
# Build Libraries AllConfigurations. This exercises the code path where we build libraries for all
```

File: eng/pipelines/global-build.yml [hunk 3]
```diff
@@ -92,6 +92,21 @@ jobs:
       buildArgs: -subset mono+libs /p:RuntimeFlavor=Mono
       timeoutInMinutes: 90
 
+#
```
Comment (danmoseley) [inline eng/pipelines/global-build.yml:95]: BTW, above it says `Build with RuntimeFlavor only. This exercise code paths where only Configuration is specified` but I think that should say `RuntimeFlavor`

**Review by danmoseley — APPROVED** [review]
Thank you for protecting not just fixing!

File: eng/pipelines/global-build.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/global-build.yml:108]: Maybe we can remove the mono legs from dev-inner loop? I know they protect the specific runtime flavor scenario, but these were added before the mobile legs existed which build that way, so maybe those are enough to protect that scenario?

**Review by safern — APPROVED** [review]

File: eng/pipelines/global-build.yml (hunk 1 above)
Comment (ViktorHofer) [reply eng/pipelines/global-build.yml:108]: @stephentoub just got the build numbers: 25m for the allconfigurations job (vs 50+ min for the other jobs in that pipeline).

> Maybe we can remove the mono legs from dev-inner loop? I know they protect the specific runtime flavor scenario, but these were added before the mobile legs existed which build that way, so maybe those are enough to protect that scenario?

@safern sounds great. Do you wanna file an issue for that / work on that?

**Review by Anipik — APPROVED** [review]

File: eng/pipelines/global-build.yml (hunk 1 above)
Comment (safern) [reply eng/pipelines/global-build.yml:108]: I filed: https://github.com/dotnet/runtime/issues/50110

**Comment by ViktorHofer** [comment]
Failure is https://github.com/dotnet/runtime/issues/48441

**Pull request merged**